var/spack/repos/builtin/packages/nix/package.py

```diff
@@ -0,0 +1,69 @@
+##############################################################################
+# Copyright (c) 2013-2017, Lawrence Livermore National Security, LLC.
+# Produced at the Lawrence Livermore National Laboratory.
+#
+# This file is part of Spack.
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/spack/spack
+# Please also see the NOTICE and LICENSE files for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+import os.path as osp
+from spack import *
+
+
+class Nix(AutotoolsPackage):
+    """Nix, the purely functional package manager"""
+
+    homepage = "http://nixos.org/nix"
+    url      = "https://github.com/NixOS/nix/archive/2.0.4.zip"
+
+    version('2.0.4', '045adeb4714f559386e391cc3c411710')
+    variant('data-dir', values=str, default=None,
+            description='Data root directory (default is /nix)')
+    variant('doc', values=bool, default=True,
+            description="Build and install documentation")
```

Comment:
this `doc` variant is failing me on bb5:

```
  GEN    doc/manual/manual.is-valid
I/O error : Attempt to load network entity http://docbook.org/xml/5.0/rng/docbook.rng
warning: failed to load external entity "http://docbook.org/xml/5.0/rng/docbook.rng"
Relax-NG parser error : xmlRelaxNGParse: could not load http://docbook.org/xml/5.0/rng/docbook.rng
Relax-NG schema http://docbook.org/xml/5.0/rng/docbook.rng failed to compile
-:1: parser error : Document is empty
make: *** [doc/manual/manual.is-valid] Error 1
make: *** Waiting for unfinished jobs....
==> Error: ProcessError: Command exited with status 2:
    'make' '-j16'
```

I  [see](
https://github.com/NixOS/nix/issues/506) people suggesting docs to be disabled. Or, there might be a fix?

Reply:
I will try with `docbook` dependency. In the meantime, let's keep this PR open.